pkg/libvirt/libvirt.go

```diff
@@ -651,12 +574,21 @@ func (d *Driver) GetIP() (string, error) {
 	if s != state.Running {
 		return "", errors.New("host is not running")
 	}
-	mac, err := d.getMAC()
+	ifaces, err := d.vm.ListAllInterfaceAddresses(libvirt.DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE)
 	if err != nil {
 		return "", err
 	}
-
-	return d.getIPByMacFromSettings(mac)
+	if len(ifaces) != 1 {
+		log.Errorf("unexpected number of interfaces for domain %s", d.MachineName)
```

Comment:
Or the list could be filtered by mac address (`const macAddress = "52:fd:fc:07:21:82"`), and then we'd error out if there is more than one matching interface?